package/MDAnalysis/analysis/pca.py

```diff
@@ -305,7 +298,7 @@ def transform(self, atomgroup, n_components=None, start=None, stop=None,
         pca_space : array, shape (number of frames, number of components)
         """
         if not self._calculated:
-            self.run()
+            self.run(start=start, stop=stop, step=step)
```

Comment:
this seems to cause errors.  Are `run` and `transform` meant to be allowed to use different frames? @kain88-de

Reply:
Yes. You may first compute the transformation and then apply it to an arbitrary trajectory to project out the motions corresponding to the PCs.

I can't think of a clean way to separate them here so instead of running `self.run()` rather raise the error and let the user do it explicitly.  Also makes for much clearer scripts on the user side.

Needs to be documented in the CHANGELOG and with a 
```reST
.. versionchanged 0.19.0:: 
   :meth:`transform` does not automatically :meth:`run` the PCA if needed; instead 
   it raises :exc:`ValueError` to remind the user to perform :meth:`run` first.
```